src/Compilers/CSharp/Test/Semantic/Semantics/PrimaryConstructorTests.cs

```diff
@@ -16717,5 +16717,862 @@ partial class Point(int i)
+                // class C2(string p2)
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p2").WithArguments("p2").WithLocation(15, 17),
+                // (19,20): error CS9105: Cannot use primary constructor parameter 'string p2' in this context.
+                //         string a = p2; 
+                Diagnostic(ErrorCode.ERR_InvalidPrimaryConstructorParameterReference, "p2").WithArguments("string p2").WithLocation(19, 20)
+                };
+
+            comp.VerifyDiagnostics(expected);
+            comp.VerifyEmitDiagnostics(expected);
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_22_InNestedType()
+        {
+            var source = @"
+class Base
+{
+    protected static string p1 = """";
+}
+
+class C1(int p1) : Base
+{
+    class Nested
+    {
+        void M()
+        {
+            string a = p1; 
+        }
+    }
+}
+
+class C2(string p2)
+{
+    class Nested
+    {
+        void M()
+        {
+            string a = p2; 
+        }
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+
+            var expected = new[] {
+                // (7,14): warning CS9113: Parameter 'p1' is unread.
+                // class C1(int p1) : Base
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p1").WithArguments("p1").WithLocation(7, 14),
+                // (18,17): warning CS9113: Parameter 'p2' is unread.
+                // class C2(string p2)
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p2").WithArguments("p2").WithLocation(18, 17),
+                // (24,24): error CS9105: Cannot use primary constructor parameter 'string p2' in this context.
+                //             string a = p2; 
+                Diagnostic(ErrorCode.ERR_InvalidPrimaryConstructorParameterReference, "p2").WithArguments("string p2").WithLocation(24, 24)
+                };
+
+            comp.VerifyDiagnostics(expected);
+            comp.VerifyEmitDiagnostics(expected);
+        }
+
+        [Fact]
+        public void ShadowedByMemberFromBase_23_InOtherConstructor()
+        {
+            var source = @"
+class Base
+{
+    protected string p1 = """";
+}
+
+class C1(int p1) : Base
+{
+    C1() : this(1)
+    {
+        string a = p1; 
+    }
+}
+
+class C2(string p2)
+{
+    C2() : this(1)
+    {
+        string a = p2; 
+    }
+}
+";
+            var comp = CreateCompilation(source, options: TestOptions.ReleaseDll);
+
+            var expected = new[] {
+                // (7,14): warning CS9113: Parameter 'p1' is unread.
+                // class C1(int p1) : Base
+                Diagnostic(ErrorCode.WRN_UnreadPrimaryConstructorParameter, "p1").WithArguments("p1").WithLocation(7, 14),
+                // (11,20): warning CS9179: Primary constructor parameter 'int p1' is shadowed by a member from base.
+                //         string a = p1; 
```

Comment:
>string a = p1;  [](http://example.com/codeflow?start=27&length=15)

Should the warning be skipped in this case since the primary constructor cannot be used here?